IRCT-RI/src/main/java/edu/harvard/hms/dbmi/bd2k/irct/ri/i2b2/I2B2XMLResourceImplementation.java

```diff
@@ -373,6 +374,49 @@ public Result runQuery(User user, Query query, Result result) throws ResourceInt
 		result.setResultStatus(ResultStatus.CREATED);
 		String projectId = "";
 
+        // gather select clauses
+        // I don't care about the performance here!!
+        // and actually this gathering select clauses block is from I2B2TranSMARTResourceImplementation.java
+        // It is working which is the only thing that I know
+		// please keep aliasMap as LinkedHashMap, because we need the sequence later
+        Map<String, String> aliasMap = new LinkedHashMap<>();
+        for (SelectClause selectClause : query
+                .getClausesOfType(SelectClause.class)) {
+            String pui = selectClause.getParameter().getPui()
+                    .replaceAll("/" + this.resourceName + "/", "");
+
+            String rawPUI = selectClause.getParameter().getPui();
```

Comment:
Absolutely not a big deal at all, but if you move this line up above the previous one, then you can use it there instead of making this same call again (String pui = rawPUI.replaceAll....)

Reply:
Agreed. But just copy paste this piece from working Transmart code, without testing it... let me change it now

Reply:
Wait a sec. I tested it, but seems this piece of code is doing the right thing.
I don't know why IntelliJ says this piece of code is useless (I've seen it before, but since the code is from a working Transmart, I just ignored it).